code/modules/mob/living/basic/jungle/seedling/seedling.dm

```diff
@@ -0,0 +1,347 @@
+		treat_hydro_tray(target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+	if(!held_can)
+		return
+
+	if(istype(target, /obj/structure/sink) || istype(target, /obj/structure/reagent_dispensers))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, target)
+		return COMPONENT_HOSTILE_NO_ATTACK
+
+
+///seedlings can water trays, remove weeds, or remove dead plants
+/mob/living/basic/seedling/proc/treat_hydro_tray(obj/machinery/hydroponics/hydro)
+
+	if(hydro.plant_status == HYDROTRAY_PLANT_DEAD)
+		balloon_alert(src, "dead plant removed")
+		hydro.set_seed(null)
+
+	if(hydro.weedlevel > 0)
+		balloon_alert(src, "weeds uprooted")
+		hydro.set_weedlevel(0)
+		return
+
+	var/list/can_reagents = held_can?.reagents.reagent_list
+
+	if(!length(can_reagents))
+		return
+
+	if((locate(/datum/reagent/water) in can_reagents) && (hydro.waterlevel < hydro.maxwater))
+		INVOKE_ASYNC(held_can, TYPE_PROC_REF(/obj/item, melee_attack_chain), src, hydro)
+		return
+
+/mob/living/basic/seedling/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+
+	if(!. || !proximity_flag || held_can)
+		return
+
+	if(!istype(attack_target, /obj/item/reagent_containers/cup/watering_can))
+		return
+
+	var/obj/item/can_target = attack_target
+	can_target.forceMove(src)
+
+/mob/living/basic/seedling/proc/change_combatant_state(state)
+	combatant_state = state
+	update_appearance()
+
+/mob/living/basic/seedling/attackby(obj/item/can, mob/living/carbon/human/user, list/modifiers)
+	if(istype(can, /obj/item/reagent_containers/cup/watering_can) && isnull(held_can))
+		can.forceMove(src)
+		return
+
+	return ..()
+
+/mob/living/basic/seedling/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(istype(arrived, /obj/item/reagent_containers/cup/watering_can))
+		held_can = arrived
+		update_appearance()
+
+	return ..()
+
+/mob/living/basic/seedling/update_overlays()
+	. = ..()
+	if(stat == DEAD)
+		. += petal_dead
+		return
+
+	switch(combatant_state)
+		if(SEEDLING_STATE_NEUTRAL)
+			. += petal_neutral
+			if(held_can)
+				. +=  mutable_appearance(icon, "seedling_can_overlay")
+		if(SEEDLING_STATE_WARMUP)
+			. += petal_warmup
+		if(SEEDLING_STATE_ACTIVE)
+			. += petal_active
+
+/mob/living/basic/seedling/update_icon_state()
+	. = ..()
+	if(stat == DEAD)
+		return
+	switch(combatant_state)
+		if(SEEDLING_STATE_NEUTRAL)
+			icon_state = "seedling"
+		if(SEEDLING_STATE_WARMUP)
+			icon_state = "seedling_charging"
+		if(SEEDLING_STATE_ACTIVE)
+			icon_state = "seedling_fire"
+
+/mob/living/basic/seedling/proc/drop_can(mob/living/user)
+	SIGNAL_HANDLER
+
+	if(isnull(held_can))
+		return
+	dropItemToGround(held_can)
+	return COMSIG_KB_ACTIVATED
+
+/mob/living/basic/seedling/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone != held_can)
+		return
+	held_can = null
+	update_appearance()
+
+/mob/living/basic/seedling/death(gibbed)
+	. = ..()
+	if(isnull(held_can))
+		return
+	held_can.forceMove(drop_location())
```

Comment:
You are also going to need a destroy proc which qdel_nulls `held_can` in case the mob is just deleted without dying.